applications/CompressiblePotentialFlowApplication/custom_response_functions/adjoint_far_field_lift_response_function.cpp

```diff
@@ -0,0 +1,275 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//
+//  Main authors:    Marc Nunez, Inigo Lopez
+//
+
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "adjoint_far_field_lift_response_function.h"
+#include "compressible_potential_flow_application_variables.h"
+#include "utilities/variable_utils.h"
+#include "utilities/parallel_utilities.h"
+
+namespace Kratos
+{
+    AdjointLiftFarFieldResponseFunction::AdjointLiftFarFieldResponseFunction(ModelPart& rModelPart, Parameters ResponseSettings)
+     : AdjointPotentialResponseFunction(rModelPart, ResponseSettings)
+    {
+
+        const Parameters default_parameters = Parameters(R"(
+        {
+            "reference_chord"             : 1.0,
+            "far_field_model_part_name"   : "",
+            "wake_normal"                 : [0,1,0],
+            "analyzer"                    : "kratos",
+            "response_type"               : "adjoint_lift_far_field",
+            "gradient_mode"               : "semi_analytic",
+            "step_size"                   : 1e-6
+        })" );
+        ResponseSettings.ValidateAndAssignDefaults(default_parameters);
+
+        KRATOS_ERROR_IF(ResponseSettings["far_field_model_part_name"].GetString()=="") << "Please define the far "
+            "field model part name in the response settings \"far_field_model_part_name\"!" << std::endl;
+        mFarFieldModelPartName = ResponseSettings["far_field_model_part_name"].GetString();
+
+        mReferenceChord = ResponseSettings["reference_chord"].GetDouble();
+        KRATOS_ERROR_IF(mReferenceChord < std::numeric_limits<double>::epsilon())
+            << "The reference chord should be larger than 0." << mReferenceChord << std::endl;
+
+        mWakeNormal = ResponseSettings["wake_normal"].GetVector();
```

Comment:
Shall we read the wake normal from the process info? I feel like if we need to define it everywhere it can lead to errors. I think it is better to just define the normal in one place and then always read it. What do you think?

Reply:
Yeah makes sense. Had to add the setting of the WAKE_NORMAL variable in both the body-fitted and embedded processes